rush-plugins/rush-buildxl-graph-plugin/src/GraphProcessor.ts

```diff
@@ -92,10 +94,10 @@ const REQUIRED_FIELDS: Array<keyof IGraphNodeInternal> = [
 /*
  * Try to get the operation id, return undefined if it fails
  */
-export function getOperationId(operation: Pick<Operation, 'associatedPhase' | 'associatedProject'>): string {
-  const task: string = operation.associatedPhase.name;
-  const project: string = operation.associatedProject.packageName;
-  return `${project}#${task}`;
+export function tryGetOperationId(operation: Partial<Operation>): string | undefined {
+  const task: string | undefined = operation.associatedPhase?.name;
+  const project: string | undefined = operation.associatedProject?.packageName;
+  return task && project ? `${project}#${task}` : undefined;
```

Comment:
Nit: use brackets to make it clearer what is being checked with the ternary operator